### DISCUSSION TITLE
How to implement two part emojis?

### DISCUSSION THREAD

**Discussion by izyumkin** [izzyumkin/EmojiPicker#10] [General]
The question remains how best to implement two part emoji.

I was doing an [issue](https://github.com/PoomSmart/EmojiLibrary/issues/26) in a repository where this logic has already been implemented. In short: Apple uses images of each skin tone for each half of the emoji.

I don't really want to store so many images in the library, so if you know an alternative solution, I will be happy to discuss it here.

For a better understanding of the context, you can read the [article](https://betterprogramming.pub/an-emoji-selection-element-aka-emojipicker-for-ios-like-in-macos-e2fa022b80af) about the library.

![4](https://user-images.githubusercontent.com/50948518/229331137-157865f1-f6ef-4d41-8a20-0702a01558ba.png)

**Comment by eemi2010** [comment]
How about storing the images somewhere in a zip file. Download and unzip the file on demand ? Then load the images from there ? This would give the ability ot decide if yes or no they want to support the feature

**Reply by izyumkin** [reply]
Just didn't want to add images to the library until last.

The idea was to draw a mask for each side of each two-part emoji via UIBezierPath. And I even made one for one. But it didn't cover the half perfectly and some pixels were showing through.

Then I read all the resources I could and realized that there is no way without images.

Already collected all the necessary materials. I'll start adding support for two-part emoji after adding search and macOS support.

**Reply by eemi2010** [reply]
Souds like a plan. Thanks for the great job.

**Reply by Jeanno** [reply]
Is the development for this feature still ongoing?
